File: src/SunAdmin.Domain/Entities/Position.cs
using SunAdmin.Domain.Common;
using SunAdmin.Domain.Enums;

namespace SunAdmin.Domain.Entities;

public sealed class Position : AuditableEntity
{
    public string Code { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public int SortOrder { get; set; }
    public RecordStatus Status { get; set; } = RecordStatus.Enabled;
    public bool IsBuiltIn { get; set; }
}

File: web/sun-admin-web/src/types/position.ts
import type { AuditableEntity, EntityStatus } from './api';

export interface PositionItem extends AuditableEntity {
  code: string;
  name: string;
  description?: string;
  sortOrder: number;
  status: EntityStatus;
  isBuiltIn?: boolean;
}

export interface PositionUpsertRequest {
  code: string;
  name: string;
  description?: string;
  sortOrder: number;
  status?: EntityStatus;
}
